### PR TITLE
MMT-1127: adds ensuring correct provider for variable manage collection associations

### PR DESCRIPTION
**Pull request by charleshuang80**
* add methods to ensure correct provider for variable manage collection associations actions
* refactor ensure correct provider methods for variables and collections
* convert ensuring correct provider for drafts to use Pundit policies

### CODE REVIEW

File: app/policies/draft_policy.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# :nodoc:
+class DraftPolicy < ApplicationPolicy
+  def show?
+    user.user.provider_id == target.provider_id
```
Comment (macrouch) [inline app/policies/draft_policy.rb]: Why do these policies use `user.user.provider_id`?

File: app/policies/draft_policy.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    # user.available_providers.include?(target.provider_id)
```
Comment (macrouch) [inline app/policies/draft_policy.rb]: commented out code

Comment (charleshuang80) [reply app/policies/draft_policy.rb]: I'll remove this.

File: app/policies/draft_policy.rb (hunk 1 above)
Comment (charleshuang80) [reply app/policies/draft_policy.rb]: We add additional user context to pass in the token (for the ACL checks), in `def pundit_user` in `ApplicationController`. That adds another level in the object, so we need to use `user.user` to access the user and `user.token` to access the token.

**Pull request merged**